src/Checkbox.js

```diff
@@ -0,0 +1,195 @@
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+`;
+
+const Input = styled(Box.withComponent('input'))`
+  clip: rect(1px, 1px, 1px, 1px);
+  height: 1px;
+  position: absolute;
+  white-space: nowrap;
+  width: 1px;
+  opacity: 0;
+  overflow: hidden;
+  padding: 0;
+`;
+
+const StyledCheckbox = ({ children, ...props }) => (
+  <Flex
+    bg="white"
+    border="1px solid"
+    borderColor="grey60"
+    borderRadius="small"
+    color="white"
+    alignItems="center"
+    justifyContent="center"
+    width={rem('16px')}
+    height={rem('16px')}
+    {...props}
+  >
+    {children}
+  </Flex>
+);
+
+const Checkbox = ({
+  id,
+  name,
+  label,
+  disabled,
+  isInvalid,
+  checked,
+  onChange,
+  value,
+  indeterminate,
+  icon,
+  ...props
+}) => (
+  <Label alignItems="center" my="regular" {...props}>
+    <Input
+      id={id}
+      type="checkbox"
+      name={name}
+      value={value}
+      checked={checked || indeterminate}
+      onChange={onChange}
+      disabled={disabled}
+      aria-invalid={isInvalid}
+    />
+    {!disabled && checked ? (
+      <StyledCheckbox
+        borderColor="blue"
+        bg={indeterminate ? 'white' : 'blue'}
+        color={indeterminate ? 'blue' : 'white'}
+      >
+        {indeterminate ? <MinusIcon /> : <CheckIcon />}
+      </StyledCheckbox>
+    ) : (
+      !disabled && <StyledCheckbox />
+    )}
+    {disabled && (
+      <StyledCheckbox
+        borderColor="grey20"
+        bg={checked ? 'grey20' : 'white'}
+        color={checked ? 'grey50' : 'white'}
+      >
+        {indeterminate ? <MinusIcon /> : <CheckIcon />}
+      </StyledCheckbox>
+    )}
+    {icon && (
+      <Icon
+        fontSize="24px"
+        name={icon}
+        color={disabled ? 'grey20' : 'grey50'}
+        ml="smaller"
+      />
+    )}
+    {label && (
+      <Text.span
+        fontSize="size2"
+        color={disabled ? 'grey40' : 'grey100'}
+        ml={icon ? 'smallest' : 'smaller'}
+      >
+        {label}
+      </Text.span>
+    )}
+  </Label>
+);
+
+Checkbox.propTypes = {
+  /**
+   * The id attribute of the checkbox.
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * The name attribute of the checkbox.
+   */
+  name: PropTypes.string,
+
+  /**
+   * Label of the checkbox.
+   */
+  label: PropTypes.node,
+
+  /**
+   * The value attribute of the checkbox.
+   */
+  value: PropTypes.string,
+
+  /**
+   * The checked attribute of the checkbox.
+   */
+  checked: PropTypes.bool,
+
+  /**
+   * State in addition to "checked" and "unchecked".
+   * When true, the checkbox displays a "minus" icon.
+   */
+  indeterminate: PropTypes.bool,
+
+  /**
+   * Function called when state changes.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * When true, the checkbox is disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * When true, the aria-invalid attribute is true.
+   * Used for accessibility.
+   */
+  isInvalid: PropTypes.bool
```

Comment:
I like the idea of support `isInvalid` for inputs in general, however, I'm having trouble understanding how a checkbox could be in an invalid state.

Reply:
Fair enough - I saw this somewhere else and just copied it over. Let's remove it.